src/semaphore.rs

```diff
@@ -228,7 +228,7 @@ impl Future for AcquireArc {
                 }
                 None => {
                     // Wait on the listener.
-                    match &mut this.listener.take() {
+                    match &mut this.listener {
                         listener @ None => {
```

Comment:
This whole block should be changed as well... doesn't this just discard the newly created event listener?

Reply:
I'm not sure what you mean. This is consistent with the non-`Arc` path and the other locks.

I do think the `listener @ None` binding can be replaced by just writing to `this.listener` to reduce some noise.

Reply:
Ah, good point.